src/task/task_environment.rs

```diff
@@ -150,11 +148,6 @@ impl<'p, D: TaskDisambiguation<'p>> SearchEnvironments<'p, D> {
                     .iter()
                     // Filter out default environment
                     .filter(|env| !env.name().is_default())
-                    // Filter out environments that can not run on this machine.
-                    .filter(|env| {
-                        self.ignore_system_requirements
```

Comment:
I think this should be added back somehow.

Reply:
I currently implemented it in such a way that it only checks the platform but this would ambigue the `cuda` vs non `cuda` tasks.... I think I'll still have to fix that.

Reply:
I tested it out, and the result is that you might get the disambigu dialog a little more often, but at least you can always go back to using `--environment` so avoid that. So I'm okay with this.